sale_cancel_reason/models/sale.py

```diff
@@ -28,8 +29,7 @@ class SaleOrder(models.Model):
     cancel_reason_id = fields.Many2one(
         'sale.order.cancel.reason',
         string="Reason for cancellation",
-        readonly=True,
-        ondelete="restrict")
```

Comment:
By removing the ondelete behaviour the default 'set null' will apply. Reasons that were applied to sale orders can be removed, leaving the value empty. Is that what you want?